src/helper_startup.py

```diff
@@ -139,16 +134,187 @@ def loadConfig():
                 os.makedirs(state.appdata)
         if not sys.platform.startswith('win'):
             os.umask(0o077)
-        BMConfigParser().save()
+        config.save()
+    else:
+        updateConfig()
 
     _loadTrustedPeer()
 
 
+def updateConfig():
+    config = BMConfigParser()
+    settingsversion = config.getint('bitmessagesettings', 'settingsversion')
+    if settingsversion == 1:
+        config.set('bitmessagesettings', 'socksproxytype', 'none')
+        config.set('bitmessagesettings', 'sockshostname', 'localhost')
+        config.set('bitmessagesettings', 'socksport', '9050')
+        config.set('bitmessagesettings', 'socksauthentication', 'false')
+        config.set('bitmessagesettings', 'socksusername', '')
+        config.set('bitmessagesettings', 'sockspassword', '')
+        config.set('bitmessagesettings', 'sockslisten', 'false')
+        config.set('bitmessagesettings', 'keysencrypted', 'false')
+        config.set('bitmessagesettings', 'messagesencrypted', 'false')
+        settingsversion = 2
+    # let class_sqlThread update SQL and continue
+    elif settingsversion == 4:
+        config.set(
+            'bitmessagesettings', 'defaultnoncetrialsperbyte',
+            str(defaults.networkDefaultProofOfWorkNonceTrialsPerByte))
+        config.set(
+            'bitmessagesettings', 'defaultpayloadlengthextrabytes',
+            str(defaults.networkDefaultPayloadLengthExtraBytes))
+        settingsversion = 5
+
+    if settingsversion == 5:
+        config.set(
+            'bitmessagesettings', 'maxacceptablenoncetrialsperbyte', '0')
+        config.set(
+            'bitmessagesettings', 'maxacceptablepayloadlengthextrabytes', '0')
+        settingsversion = 7
+
+    # Raise the default required difficulty from 1 to 2
```

Comment:
Remove this commented code if not required.

Reply:
I retained commented out code in order to facilitate a review, so you can see how I interpret logic of the code. I can remove any commented code by final commit if all reviewers agree that changes are correct and the code is indeed not required.

Reply:
ok, I think If we write a function with in 15 - 20 lines its better to have a good review.

Reply:
I'd be more in favour of using the PR description, self-comments on the PR or a gist, referred to in the PR description rather than removing comments about the change in a final, post-review commit. Comments should be about the code in a particular revision, not the change/diff.

So, rather than saying `# increase frobulation from 1 to 3` there should be a `# frobulation should be 1 due to the small size of the userbase` comment in the earlier revision and `# frobulation should be 3 due to the large size of the userbase` in the later revision. Viewing the diff should provide the transitional view that the frobulation has changed from 1 to 3.

But I don't have particularly strong feelings about this.